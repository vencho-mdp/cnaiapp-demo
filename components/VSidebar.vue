<template>
  <div
    data-test="sidebar"
    class="bg-black-light flex min-h-screen w-screen top-0 left-0 absolute justify-end"
  >
    <aside
      :class="
        'bg-primary-darkblue flex flex-col h-screen max-h-screen w-80 overflow-auto' +
        ' ' +
        sidebarClasses
      "
    >
      <button class="flex items-center justify-end">
        <img
          class="cursor-pointer h-6 m-4 w-6"
          src="~/assets/images/close-icon.svg"
          alt="Cerrar"
          @click="$emit('closeSidebar')"
        />
      </button>
      <h1
        class="text-base font-bold text-center text-white p-0.5"
        :class="applyTextMargin ? 'mt-8' : null"
      >
        {{ title }}
      </h1>
      <h2
        v-if="subtitle"
        class="text-white text-xs font-bold mt-8 text-center px-1"
      >
        {{ subtitle }}
      </h2>
      <slot name="content" />
    </aside>
  </div>
</template>

<script>
export default {
  props: {
    title: {
      type: String,
      default: "",
    },
    subtitle: {
      type: String,
      default: "",
    },
    applyTextMargin: {
      type: Boolean,
      default: true,
    },

    remove_overflow_delay: {
      type: Number,
      default: 750,
    },
    sidebarClasses: {
      type: String,
      default: "",
    },
  },
  mounted() {
    window.scrollTo({ top: 0, behavior: "smooth" });
  },
  beforeCreate() {
    if (process.browser) {
      document.getElementsByTagName("body")[0].classList.add("overflow-hidden");
    }
  },
  beforeDestroy() {
    // close  sidebar first
    setTimeout(() => {
      document
        .getElementsByTagName("body")[0]
        .classList.remove("overflow-hidden");
    }, this.remove_overflow_delay);
  },
};
</script>
