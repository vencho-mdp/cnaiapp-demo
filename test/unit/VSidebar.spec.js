import { mount } from '@vue/test-utils'
import VSidebar from '@/components/VSidebar.vue'

describe('VSidebar', () => {
  window.scrollTo = jest.fn()
  it('renders correctly', async () => {
    const el = mount(VSidebar)
    expect(el.html()).toMatchSnapshot()
  })

  it('should render correctly with slots', async () => {
    const el = mount(VSidebar, {
      slots: {
        content: '<div> Content Example </div>'
      }
    })
    const text = el.find('div').text()
    expect(text).toBe('Content Example')
  })

  it('should render correctly with content props', async () => {
    const el = mount(VSidebar, {
      propsData: {
        title: 'Title Example',
        subtitle: 'Subtitle Example'
      }
    })
    const h1 = el.find('h1').text()
    expect(h1).toBe('Title Example')
    const h2 = el.find('h2').text()
    expect(h2).toBe('Subtitle Example')
  })
})
