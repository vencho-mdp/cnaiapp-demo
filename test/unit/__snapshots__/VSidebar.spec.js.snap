// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VSidebar renders correctly 1`] = `
"<div data-test=\\"sidebar\\" class=\\"bg-black-light flex min-h-screen w-screen top-0 left-0 absolute justify-end\\">
  <aside class=\\"bg-primary-darkblue flex flex-col h-screen max-h-screen w-80 overflow-auto\\"><button class=\\"flex items-center justify-end\\"><img src=\\"~/assets/images/close-icon.svg\\" alt=\\"Cerrar\\" class=\\"cursor-pointer h-6 m-4 w-6\\"></button>
    <h1 class=\\"text-base font-bold mt-8 text-center text-white\\">

    </h1>
    <!---->
  </aside>
</div>"
`;
